plugin/trino-hive/src/main/java/io/trino/plugin/hive/DefaultHiveViewReaderFactory.java

```diff
@@ -0,0 +1,72 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.inject.Inject;
+import io.trino.plugin.hive.metastore.CoralSemiTransactionalHiveMSCAdapter;
+import io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.MetadataProvider;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.type.TypeManager;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import static io.trino.plugin.hive.HiveSessionProperties.isHiveViewsLegacyTranslation;
+import static io.trino.plugin.hive.ViewReaderUtil.HiveViewReader;
+import static io.trino.plugin.hive.ViewReaderUtil.PrestoViewReader;
+import static io.trino.plugin.hive.ViewReaderUtil.coralTableRedirectionResolver;
+import static io.trino.plugin.hive.ViewReaderUtil.isTrinoView;
+import static java.util.Objects.requireNonNull;
+
+public class DefaultHiveViewReaderFactory
+        implements HiveViewReaderFactory
+{
+    private final TypeManager typeManager;
+    private final MetadataProvider metadataProvider;
+    private final HiveTimestampPrecision hiveViewsTimestampPrecision;
+    private final boolean runHiveViewRunAsInvoker;
+
+    @Inject
+    public DefaultHiveViewReaderFactory(
+            TypeManager typeManager,
+            MetadataProvider metadataProvider,
+            HiveConfig hiveConfig)
+    {
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.metadataProvider = requireNonNull(metadataProvider, "metadataProvider is null");
+        this.hiveViewsTimestampPrecision = hiveConfig.getTimestampPrecision();
+        this.runHiveViewRunAsInvoker = hiveConfig.isHiveViewsRunAsInvoker();
+    }
+
+    @Override
+    public ViewReader createViewReader(SemiTransactionalHiveMetastore metastore, ConnectorSession session, Table table, BiFunction<ConnectorSession, SchemaTableName, Optional<CatalogSchemaTableName>> tableRedirectionResolver)
+    {
+        if (isTrinoView(table)) {
+            return new PrestoViewReader();
+        }
+        if (isHiveViewsLegacyTranslation(session)) {
+            return new LegacyHiveViewReader(runHiveViewRunAsInvoker);
+        }
+
+        return new HiveViewReader(
+                new CoralSemiTransactionalHiveMSCAdapter(metastore, coralTableRedirectionResolver(session, tableRedirectionResolver, metadataProvider)),
+                typeManager,
+                runHiveViewRunAsInvoker,
+                hiveViewsTimestampPrecision);
```

Comment:
Can we compute the precision from the session i.e `HiveSessionProperties#getTimestampPrecision`

Reply:
previously this value was take from hive config.
If we change it to `HiveSessionProperties#getTimestampPrecision` we have some test failures.

Reply:
so i reverted to previous default behaviour